### PR TITLE
fix(data-classes): docstring typos and clean up

### PR DESCRIPTION
**Pull request by michaelbrewer**
**Issue #, if available:**

It's been a while since i scanned through the data classes for typos and improvements.

## Description of changes:

Changes:
- Use correct wording: `built-in`, `its`, `sign in`, `all the` and `case-sensitive`
- Modern use of `super()`
- Use `resource.lstrip("/")` instead of `resource[:1] == "/"` for readability to remove leading "/" that isn't allowed when constructing an authorizer response object

**Checklist**

 

* [X] [Meet tenets criteria](https://awslabs.github.io/aws-lambda-powertools-python/#tenets)
* [X] Update tests
* [X] Update docs
* [X] PR title follows [conventional commit semantics](https://github.com/awslabs/aws-lambda-powertools-python/blob/376ec0a2ac0d2a40e0af5717bef42ff84ca0d1b9/.github/semantic.yml#L2)

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by michaelbrewer** [comment]
@heitorlessa  - some small housekeeping fixes.

**Review by heitorlessa — CHANGES_REQUESTED** [review]
one question on whether you meant `startswith` vs `==`, as I can't remember what the possible values can be.

File: aws_lambda_powertools/utilities/data_classes/api_gateway_authorizer_event.py [hunk 1]
```diff
@@ -440,7 +440,7 @@ def _add_route(self, effect: str, http_method: str, resource: str, conditions: O
         if not self._resource_pattern.match(resource):
             raise ValueError(f"Invalid resource path: {resource}. Path should match {self.path_regex}")
 
-        if resource[:1] == "/":
+        if resource.startswith("/"):
```
[the remark this answers is not in the document]
Comment (michaelbrewer) [reply aws_lambda_powertools/utilities/data_classes/api_gateway_authorizer_event.py:443]: for readability. Later this can be replaced with `removeprefix`

**Review by heitorlessa — CHANGES_REQUESTED** [review]
Thanks for the changes - one last one, please remove the empty `logfile`, as it seems to have been added by accident.

File: aws_lambda_powertools/utilities/data_classes/api_gateway_authorizer_event.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (heitorlessa) [reply aws_lambda_powertools/utilities/data_classes/api_gateway_authorizer_event.py]: If you can add an one-line comment on that resource line it'd be great to help onboard new people to it.

[the remark this answers is not in the document]
Comment (michaelbrewer) [reply aws_lambda_powertools/utilities/data_classes/api_gateway_authorizer_event.py]: @heitorlessa tried anoher one

[the remark this answers is not in the document]
Comment (michaelbrewer) [reply aws_lambda_powertools/utilities/data_classes/api_gateway_authorizer_event.py]: @heitorlessa - it is using `lstrip` now which is simple and fast

**Review by heitorlessa — APPROVED** [review]

**Comment by heitorlessa** [comment]
Updated description on `.lstrip` to clarify what this is for for anyone looking at this in the future.

**Pull request merged**